### PR TITLE
Adds endpoint for non-revocable session token upgrade

### LINKED ISSUE #2255: {"code":209,"error":"invalid session token"} on v2.2.15
I upgraded parse-server from v2.2.7 to v2.2.15. I started getting {"code":209,"error":"invalid session token"} for all the users who had logged in 2-3 days earlier. when i reverted back to v2.2.7. things started working again. is there an issue with v2.2.15

### PR DESCRIPTION
**Pull request by flovilmart**
Better support for upgrading to revocable session after migration has occurred.

Fixes #2255

### CODE REVIEW

**Comment by facebook-github-bot-7** [comment]
@flovilmart updated the pull request - [view changes](https://github.com/ParsePlatform/parse-server/pull/2646/files/e855c1ae276f27f08c9fc5b950c24103ee07ec04..0138070d431dc8301eabeb757a8893c0108565df)

**Comment by facebook-github-bot-9** [comment]
@flovilmart updated the pull request - [view changes](https://github.com/ParsePlatform/parse-server/pull/2646/files/0138070d431dc8301eabeb757a8893c0108565df..e855c1ae276f27f08c9fc5b950c24103ee07ec04)

File: src/middlewares.js [hunk 1]
```diff
@@ -147,8 +147,16 @@ export function handleParseHeaders(req, res, next) {
     return;
   }
 
-  return auth.getAuthForSessionToken({ config: req.config, installationId: info.installationId, sessionToken: info.sessionToken })
-    .then((auth) => {
+  return Promise.resolve().then(() => {
+    // handle the upgradeToRevocableSession path on it's own
+    if (info.sessionToken && 
+        req.url === '/upgradeToRevocableSession' && 
+        info.sessionToken.indexOf('r:') != 0) {
```
Comment (steven-supersolid) [inline src/middlewares.js]: Doesn't the presence of r: in the session token indicate non-legacy?

Comment (flovilmart) [inline src/middlewares.js]: Yes, that's why != 0 as if it's revocable, it's gonna be at that index.

Comment (steven-supersolid) [inline src/middlewares.js]: Ah doh, I misread :)

**Pull request merged**